uptane-standard.md

```diff
@@ -289,20 +289,309 @@ The Timestamp role SHALL produce and sign metadata indicating whether there are
 
 ## Metadata abstract syntax {#meta_syntax}
 
+Metadata files on a repository SHOULD be written using the ABNF abstract syntax specified in this section.  These files MAY be encoded and decoded using any transfer syntax that an OEM desires (e.g., BER, CER, DER, JSON, OER, PER, XER).
+
 ### Common Metadata Structures and Formats
 
+Metadata files SHOULD share the data structures in this section. These data structures specify how information, such as cryptographic hashes, digital signatures, and public keys, should be encoded.
+
+This is an ABNF module that defines common data structures used by metadata files.
+
+```
+RoleType        = "root" / "targets" / "snapshot" / "timestamp"
+
+; String types.
+Filename        = 1*32VCHAR
+; No known path separator allowed in a strict filename.
+StrictFilename  = %x21-2E / %x30-5B / %x5D-7E
+BitString       = 1*1024BIT
+OctetString     = 1*1024OCTET
+HexString       = 1*1024HEXDIG
+; Table 1 of RFC 4648.
+Base64String    = 1*1024(ALPHA / DIGIT / "+" / "=" / "/")
+; Adjust length to your needs.
+Paths           = 1*8(Path)
+Path            = 1*32((ALPHA / "_" / "*" / "\"" / "/"))
+; Adjust length to your needs.
+URLs            = *8URL
+URL             = 1*1024VCHAR
+; A generic identifier for vehicles, primaries, secondaries.
+Identifier      = 1*32VCHAR
+
+Natural         = *DIGIT
+Positive        = *("1" / "2" / "3" / "4" / "5" / "6" / "7" / "8" / "9")
+Length          = Positive
+Threshold       = Positive
+Version         = Positive
+; The date and time in UTC encoded as a UNIX timestamp.
+UTCDateTime     = Positive
+
+BinaryData      = BitString / OctetString / HexString / Base64String
+
+; Adjust length to your needs.
+Hashes          = 1*8Hash
+Hash            = HashFunction  BinaryData
+
+HashFunction    = "sha224" / "sha256" / "sha384" / "sha512" / "sha512-224" /
+                               "sha512-256" / ...
+
+; Adjust length to your needs.
+Keyids          = 1*8Keyid
+; Usually, a hash of a public key.
+Keyid           = HexString
+
+; Adjust length to your needs.
+Signatures      = 1*8Signature
+Signature       = Keyid SignatureMethod Hash HexString
+SignatureMethod = "rsassa-pss" / "ed25519" / ...
```

Comment:
We might want to bring this up to date with TAP 9, which mitigates an important security problem